bson/src/main/org/bson/codecs/NumberCodecHelper.java

```diff
@@ -45,6 +48,13 @@ static int decodeInt(final BsonReader reader) {
                     throw invalidConversion(Integer.class, doubleValue);
                 }
                 break;
+            case DECIMAL128:
```

Comment:
This class is used by Long/Integer/Double/Float Codecs.  Since Decimal128 now is-a Number, it makes sense to support decoding from Decimal128 to any of these types.  It's pretty restrictive, since it throws if we lose precision (see the tests), but we could relax it in the future if there's demand.

Reply:
Seems fair we currently check correctness for all conversions.

Reply:
Yup